pkgs/servers/nfd/default.nix

```
@@ -0,0 +1,53 @@
+{ lib
+, stdenv
+, boost
+, fetchFromGitHub
+, libpcap
+, ndn-cxx
+, openssl
+, pkg-config
+, sphinx
+, systemd
+, wafHook
+, websocketpp
+, withSystemd ? stdenv.isLinux
+, withWebSocket ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nfd";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "named-data";
+    repo = lib.toUpper pname;
+    rev = "NFD-${version}";
+    sha256 = "1l9bchj8c68r6qw4vr1kc96jgxl0vpqa2vjkvy1xmhz92sivr6gi";
+    fetchSubmodules = withWebSocket;
+  };
+
+  nativeBuildInputs = [ pkg-config sphinx wafHook ];
+  buildInputs = [ libpcap ndn-cxx openssl websocketpp ] ++ lib.optional withSystemd systemd;
+
+  wafConfigureFlags = [
+    "--boost-includes=${boost.dev}/include"
+    "--boost-libs=${boost.out}/lib"
+    "--with-tests"
+  ] ++ lib.optional (!withWebSocket) "--without-websocket";
```

```suggestion
    (lib.withFeature withWebSocket "websocket")
  ];
```

Providing the flag unconditionally avoids any worries about whether websocket is enabled or disabled by default by the configuration

---

I didn't know of the `withFeature` function. Nice to have, I'll update the code.

---

I just checked for this suggestion and unfortunately the build script won't work with a `--with-websocket` argument: it only checks for the without version and defaults to including it in the build.
A more pleasant version of the solution I hardcoded is with `lib.optionalString`